### PR TITLE
Basic authentication for state/county admins

### PR DESCRIPTION
**Pull request by dmzimmerman**

### CODE REVIEW

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/Authentication.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  
+  /**
+   * Checks to see whether the specified request is authenticated as
+   * the specified administrator type.
+   * 
+   * @param the_request The request.
+   * @param the_type The administrator type.
+   * @return true if the request is authenticated, false otherwise.
+   */
+  public static boolean isAuthenticated(final Request the_request,
```
Comment (ranweiler) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/Authentication.java]: Maybe rename this to `isAuthenticatedAs()`, which would match the `authenticateAs()` method below.

Comment (dmzimmerman) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/Authentication.java]: Seems like a reasonable change.

**Comment by dmzimmerman** [comment]
Yeah... I can't really do much more until we have some idea of how we're talking to two-factor authentication and the like. The basic bones of it in terms of how it's tracked within a session won't likely change much, though probably a sanity check that "this is a county administrator for the county they are trying to administer" will need to be added soon.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/Authentication.java (hunk 1 above)
Comment (dmzimmerman) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/Authentication.java]: f3dd22d

**Comment by ranweiler** [comment]
@dmzimmerman, the new commits look good! :ship:

**Pull request merged**